homeassistant/components/knx/__init__.py

```diff
@@ -229,6 +230,19 @@ async def async_setup(hass, config):
             "https://www.home-assistant.io/blog/2020/09/17/release-115/#breaking-changes"
         )
 
+    # deprecation warning since 2021.3
+    if CONF_KNX_CONFIG in config[DOMAIN]:
+        config_file_deprecation_warning = (
+            f"The 'config_file' option will soon be deprecated. Please replace it with Home Assistant config schema "
+            f"directly in `configuration.yaml` (see https://www.home-assistant.io/integrations/knx/). \n"
+            f"An online configuration converter tool for your `{config[DOMAIN][CONF_KNX_CONFIG]}` is available at https://xknx.io/config-converter/"
+        )
+        _LOGGER.warning(config_file_deprecation_warning)
+        hass.components.persistent_notification.async_create(
+            config_file_deprecation_warning,
+            title="KNX",
+        )
```

Comment:
We really should not send a persistent notification for this. We never do this, and we should not introduce it.

Reply:
OK. Shall I now add `cv.deprecated` again, or just log my message to warning?

Reply:
A custom deprecation notice is acceptable, `cv.deprecated` is preferred in general. It was merely the persistent notification my comment was about.